packages/mui-styled-engine/src/StyledEngineProvider/StyledEngineProvider.js

```diff
@@ -98,10 +101,15 @@ function getCache(injectFirst, enableCssLayer) {
 
 export default function StyledEngineProvider(props) {
   const { injectFirst, enableCssLayer, children } = props;
-  const cache = React.useMemo(
-    () => getCache(injectFirst, enableCssLayer),
-    [injectFirst, enableCssLayer],
-  );
+  const cache = React.useMemo(() => {
+    const cacheKey = `${injectFirst}-${enableCssLayer}`;
+    if (cacheMap.has(cacheKey)) {
```

Comment:
FYI, I'm seeing an issue from this change in SSR where emotion thinks a node has already been inserted (because it was on a previous request that is now using a shared cache).

The fix I have patched into my repo is to change this if check to

```ts
    if (typeof document === 'object' && cacheMap.has(cacheKey)) {
```

Reply:
opened PR here: https://github.com/mui/material-ui/pull/46276